include/deal.II/lac/cuda_vector.h

```diff
@@ -298,7 +311,7 @@ namespace LinearAlgebra
       /**
        * Pointer to the array of elements of this vector.
        */
-      Number *val;
+      std::unique_ptr<Number, void (*)(Number *)> val;
```

Comment:
that's the wrong type. for arrays, you need to use `std::unique_ptr<Number[],...>`

Reply:
actually, it probably doesn't make a difference since you have a custom deleter anyway. But I'd think it's the right thing to do.